packages/app/src/cli/services/deploy/upload.ts

```diff
@@ -240,6 +237,27 @@ ${output.token.json(userErrors)}
   return res.data.functionSet.function?.id as string
 }
 
+async function uploadWasmBlob(extension: FunctionExtension, apiKey: string, token: string): Promise<string> {
+  const {url, headers, maxSize} = await getFunctionExtensionUploadURL({apiKey, token})
+  headers['Content-Type'] = 'application/wasm'
+
+  const functionContent = fs.readFileSync(extension.buildWasmPath())
+  const res = await http.fetch(url, {body: functionContent, headers, method: 'PUT'})
+  const resBody = res.body?.read()?.toString() || ''
+
+  if (res.status === 200) {
+    return url
+  } else if (res.status === 400 && resBody.includes('EntityTooLarge')) {
+    const errorMessage = output.content`The size of the Wasm binary file for Function ${extension.localIdentifier} is too large. It must be less than ${maxSize}.`
+    throw new error.Abort(errorMessage)
+  } else {
+    const errorMessage = output.content`Something went wrong uploading the Function ${
+      extension.localIdentifier
+    }. The server responded with status ${res.status.toString()} and body: ${resBody}`
+    throw new error.Bug(errorMessage)
+  }
```

Comment:
I'd be more granular here. Considering 5xx errors are already tracked server-side, we don't want them to show up in the CLI Bugsnag project. I'd only report as bugs those that are 4xx and that are not caused by the binary size exceeded.